tests/test_signed_payload.py

```diff
@@ -3,74 +3,120 @@
 import json
 import base64
 
-
 class TestApi(TestCase):
-    def print_payload_signature(self):
+    def sign_message(self, message):
         pk = keys.PrivateKey(b'\x01' * 32)
+        signature = pk.sign_msg(message)
+        signature_bytes =signature.to_bytes()
+        public_address = pk.public_key.to_checksum_address()
+        return signature_bytes, public_address
 
-        payload = {
-            "service_proposal": {
-                "id": 1,
-                "format": "service-proposal/v1",
-                "provider_id": "node1",
-            }
+    def test_incorrect_authorization_header(self):
+        headers = {
+            "incorrectAuthorizationHeader": "Signature sig"
         }
 
-        signature = pk.sign_msg(json.dumps(payload))
-        print json.dumps(payload)
-        print pk.public_key.to_checksum_address().lower()
-        print base64.b64encode(signature.to_bytes())
+        re = self.client.post(
+            '/v1/test_signed_payload',
+            data=json.dumps(''),
+            headers=headers
+        )
 
-    def test_signed_payload(self):
-        payload = {
-            "service_proposal": {
-                "id": 1,
-                "format": "service-proposal/v1",
-                "provider_id": "node1",
-            }
+        self.assertEqual(401, re.status_code)
+        self.assertEqual(re.json['error'], 'missing Authorization in request header')
+
+    def test_incorrect_authorization_header_value_format(self):
+        headers = {
+            "Authorization": "Signature"
         }
 
+        re = self.client.post(
+            '/v1/test_signed_payload',
+            data=json.dumps(''),
+            headers=headers
+        )
+
+        self.assertEqual(401, re.status_code)
+        self.assertEqual(re.json['error'], 'invalid Authorization header value provided, correct format: Signature <signature_base64_encoded>')
+
+    def test_incorrect_authorization_type(self):
         headers = {
-            "identity": "0x1a642f0e3c3af545e7acbd38b07251b3990914f1",
-            "Authorization": "Signature Da1mAwK5abmXQCNsCE+YjsZbR9jTyEKqdrjxxMKwNzwr2NFnM35UiVQJWcg8rgL+X2PR60LoIUMlGU9OPaSoZwE="
+            "Authorization": "incorrectType sig"
         }
 
         re = self.client.post(
-            '/v1/signed_payload',
-            data=json.dumps(payload),
+            '/v1/test_signed_payload',
+            data=json.dumps(''),
             headers=headers
         )
 
-        self.assertEqual({}, re.json)
-        self.assertEqual(200, re.status_code)
+        self.assertEqual(401, re.status_code)
+        self.assertEqual(re.json['error'], 'authentication type have to be Signature')
 
-    def test_incorrectly_signed_payload(self):
-        payload = {}
+    def test_empty_authorization_header_value(self):
+        headers = {
+            "Authorization": "Signature "
+        }
+
+        re = self.client.post(
+            '/v1/test_signed_payload',
+            data=json.dumps(''),
```

Comment:
If we remove `@validate_json` from `/v1/test_signed_payload`, we wouldn't have to pass in an empty post data - this endpoint shouldn't require any post data.